sklearn/utils/tests/test_parallel.py

```diff
@@ -17,8 +17,9 @@ def test_configuration_passes_through_to_joblib(n_jobs, backend):
     # Tests that the global global configuration is passed to joblib jobs
 
     with config_context(working_memory=123):
-        results = Parallel(n_jobs=n_jobs, backend=backend)(
-            delayed(get_working_memory)() for _ in range(2)
+        config = get_config()
+        results = Parallel(n_jobs=n_jobs, backend=backend, pre_dispatch=1)(
+            delayed(get_working_memory, config=config)() for _ in range(10)
         )
 
-    assert_array_equal(results, [123] * 2)
+    assert_array_equal(results, [123] * 10)
```

Comment:
Maybe you could extend this test to show that this patterns also work from other threads, e.g. with something like the following (untested):

```python
    results = []

    def parallel_inspect_config():
        with sklearn.config_context(working_memory=123):
            config = sklearn.get_config()
            results.extend(
                Parallel(n_jobs=2, pre_dispatch=4)(
                    delayed(get_working_memory, config=config)() for _ in range(n_iter)
                )
            )

    other_thread = threading.Thread(target=parallel_inspect_config)
    other_thread.start()
    other_thread.join()

    assert results == [123] * n_iter
```

It would even be better to have a test with ThreadpoolExecutor that checks that concurrently running threads calling joblib parallel with different contexts do not result in mixed up configurations.

Reply:
I added this test.